shell/platform/darwin/common/string_conversions.mm

```diff
@@ -6,20 +6,33 @@
 
 namespace shell {
 
-std::vector<uint8_t> GetVectorFromNSString(NSString* string) {
-  if (!string.length)
+NSData* GetNSDataFromNSString(NSString* string) {
+  if (!string)
```

Comment:
This can't be right. You are just checking if string is not `nil`. All valid strings will give you `nil` `NSData` instances.

Once you do fix the logic, you can write this as `[string length] == 0`. If `string` is `nil`, you will end up sending a message to `nil` which is fine. If it is not `nil`, you will do the length check.